pkg/adaptation/suite_test.go

```diff
@@ -93,9 +98,11 @@ func (s *Suite) Dir() string {
 
 // Startup starts up the test suite.
 func (s *Suite) Startup() {
+	plugins := s.plugins
+	s.plugins = nil
```

Comment:
Why do we need to clear the plugins and re-append them in `StartPlugins()`?

Reply:
That's because I changed Suite.StartPlugins() and Suite.WaitForPluginsToSync so that instead of starting up/waiting for Suite.plugins they takes the plugins to start/wait for as arguments. This simplifies a bit the code for some of the latest test cases additions, where we first start the suite with some plugins, and then as part of the test case we we later start some more.

But I kept Suite.Prepare() intact which stores the plugins to start during Suite.Startup() in Suite.plugins. Then, to not end up with the initially started plugins twice in Suite.plugins, the simplest change was to reset the slice in Startup() and let StartPlugins() repopulate it.